### PR TITLE
pipeline: add and integrate encryption pipeline

### PR DESCRIPTION
**Pull request by acud**
Adds an encryption pipeline. No integration.

### CODE REVIEW

**Comment by janos** [comment]
This PR needs conflicts to be resolved.

File: pkg/file/pipeline/encryption_writer.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+func newSpanEncryption(key encryption.Key) *encryption.Encryption {
+	refSize := int64(swarm.HashSize + encryption.KeyLength)
+	return encryption.New(key, 0, uint32(swarm.ChunkSize/refSize), sha3.NewLegacyKeccak256)
+}
+
+func newDataEncryption(key encryption.Key) *encryption.Encryption {
+	return encryption.New(key, int(swarm.ChunkSize), 0, sha3.NewLegacyKeccak256)
+}
```
Comment (janos) [inline pkg/file/pipeline/encryption_writer.go:57]: This makes encryption writer coupled with the encryption package implementation. In order to have unit tests here independent on the encryption implementation, and possible to simplify them, encrytion.Encryptor should be a dependency of encryptionWriter.

**Review by janos — APPROVED** [review]

File: pkg/file/pipeline/encryption_writer.go (hunk 1 above)
Comment (acud) [reply pkg/file/pipeline/encryption_writer.go:57]: can we have it in this way and i'll pull it out? it probably needs better crafting, and together with the mock encryption i can build unit tests properly

Comment (janos) [reply pkg/file/pipeline/encryption_writer.go:57]: Yes, let's address this later.

**Review by zelig — APPROVED** [review]

**Review by zbiljic — APPROVED** [review]

**Pull request merged**